sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_receiver_async.py

```diff
@@ -826,6 +828,79 @@ async def peek_messages(
         ):
             return messages
 
+    async def delete_messages(
+        self,
+        *,
+        max_message_count: Optional[int] = None,
+        before_enqueued_time_utc: Optional[datetime.datetime] = None,
+    ) -> int:
+        """  This operation deletes messages in the queue that are older than the specified enqueued time,
+         up to 4,000 messages at a time.
+
+        :param int or None max_message_count: The maximum number of messages to delete. The default value is 1.
+        :param datetime.datetime or None before_enqueued_time_utc: The UTC datetime value before which all messages
+         should be deleted.The default value is None, meaning all messages in the queue will be considered.
+        :rtype: int
+
+        """
+        self._check_live()
+        await self._open()
+
+        message_count = max_message_count if max_message_count else 4000
```

Comment:
we should raise a `ValueError` here if `max_message_count` is greater than 4000. Will the service return an error instead?

Reply:
service returns a 500 error currently but have pointed that out to them

Reply:
does it contain detail that would let the user know why it failed?

Reply:
It does not it is also not advised to raise a 500, the suggestion was to remove it and raise a better error

Reply:
great, would be preferable to raise the error from the service itself jic the limit changes in the future